docs/api/platform/MpuManagement.md

```diff
@@ -0,0 +1,47 @@
+## MPU management
+
+Memory protection for Mbed OS is enabled automatically for devices that support the MPU API. The MPU management functions provided here allow libraries and applications to turn off these memory protections if necessary. The memory protection the MPU provides does the following:
+
+- Prevents execution from RAM.
+- Prevents writing to ROM.
+
+Mbed OS handles MPU management automatically in the following situations:
+
+- Memory protection is enabled as part of the boot sequence.
+- Memory protection is disabled when starting a new application.
+- Memory protection is disabled while flash programming.
+
+<span class="notes">**Note:** Memory protection should be transparent to most applications and libraries because Mbed OS handles it automatically for operations that need to disable MPU protections, such as flash programming.</span>
```

Comment:
Maybe add "this is an advanced feature, most users should not have to worry about this - only use it if you know what you're doing" type of warning?

Reply:
@AnotherButler had concerning about the phrasing here. I'll let her make this change.